src/test/java/com/microsoft/sqlserver/jdbc/fedauth/PooledConnectionTest.java

```diff
@@ -0,0 +1,294 @@
+                                            || e.getMessage().contains(ERR_MSG_HAS_CLOSED)
+                                            || e.getMessage().contains(ERR_MSG_HAS_BEEN_CLOSED)
+                                            || e.getMessage().contains(ERR_MSG_SOCKET_CLOSED));
+                        }
+                    }
+                }.start();
+            }
+
+            // sleep in order to catch exception from other threads if tests fail.
+            try {
+                Thread.sleep(TimeUnit.SECONDS.toMillis(60));
+
+            } catch (InterruptedException e) {
+                fail(e.getMessage());
+            }
+        } catch (Exception e) {
+            fail(e.getMessage());
+        }
+    }
+
+    // suspend until access token expires
+    @Test
+    public void testPooledConnectionWithAccessToken() throws SQLException {
+        try {
+            SQLServerConnectionPoolDataSource ds = new SQLServerConnectionPoolDataSource();
+
+            ds.setServerName(azureServer);
+            ds.setDatabaseName(azureDatabase);
+            ds.setAccessToken(accessToken);
+
+            // create pooled connection
+            final PooledConnection pc = ds.getPooledConnection();
+
+            // get first connection from pool
+            try (Connection connection1 = pc.getConnection()) {
+                testUserName(connection1, azureUserName, SqlAuthentication.NotSpecified);
+            }
+
+            Random rand = new Random();
+            int numberOfThreadsForEachType = rand.nextInt(15) + 1; // 1 to 15
+            for (int i = 0; i < numberOfThreadsForEachType; i++) {
+                new Thread() {
+                    public void run() {
+                        try (Connection connection2 = pc.getConnection()) {
+                            testUserName(connection2, azureUserName, SqlAuthentication.NotSpecified);
+                        } catch (SQLException e) {
+                            fail(e.getMessage());
+                        }
+                    }
+                }.start();
+
+                new Thread() {
+                    public void run() {
+                        try (Connection connection2 = pc.getConnection()) {
+                            testUserName(connection2, azureUserName, SqlAuthentication.NotSpecified);
+                        } catch (SQLException e) {
+                            fail(e.getMessage());
+                        }
+                    }
+                }.start();
+
+                new Thread() {
+                    public void run() {
+                        try (Connection connection2 = pc.getConnection()) {
+                            testUserName(connection2, azureUserName, SqlAuthentication.NotSpecified);
+                        } catch (SQLException e) {
+                            fail(e.getMessage());
+                        }
+                    }
+                }.start();
+            }

```

Comment:
Similar idea here, just create a runnable out of the `run()` and then pass it to the threads.